### PR TITLE
Adding typings file to fix #3916

### LINKED ISSUE #3916: swiper/vue missing type declarations
This is a (multiple allowed):
* [ ] bug
* [x] enhancement
* [ ] feature-discussion (RFC)

I see that Swiper has already types for `swiper/react` component included. I would be really cool to add type definitions for `swiper/vue` component as well. This would allow us to integrate swiper in vue.js projects with typescript easier.

Screenshot 2020-11-03 at 8 35 50

I'm willing to work on this feature with a little guidance from maintainers.

* Swiper Version: v3.6.5
* Platform/Target and Browser Versions: Typescript

### PR DESCRIPTION
**Pull request by GerryWilko**
This initial typings file will allow the package to be used with Vue Typescript projects and use the Swiper and SwiperSlide as components.

Fixes #3916

### CODE REVIEW

**Comment by nolimits4web** [comment]
Is it compatible with Vue.js v3 ?

**Comment by GerryWilko** [comment]
Yeah should be because it doesn't really care what the NPM Vue library is. Does swiper's Vue library support v3 though? The api for components has changed significantly and its not backwards compatible.

**Comment by nolimits4web** [comment]
Swiper compatible ONLY with Vue 3. Seems like these typings are not valid for Vue 3, because Vue 3 doesn't have default export

**Comment by GerryWilko** [comment]
I use this typings file with Vue 2 so Swiper is compatible with Vue 2. Let me do some investigating and see how I can make it work for both.

**Comment by vltansky** [comment]
@GerryWilko hey, any updates?

File: src/swiper-vue.d.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+declare module 'swiper/vue' {
+  import _Vue from 'vue';
+
+  export class Swiper extends _Vue {}
+
+  export class SwiperSlide extends _Vue {}
```
Comment (pzi) [inline src/swiper-vue.d.ts:6]:
```suggestion
  export class SwiperZ extends _Vue {}

  export class SwiperSlide extends _Vue {}
```

**Comment by vltansky** [comment]
types added here https://github.com/nolimits4web/swiper/pull/4707?notification_referrer_id=MDE4Ok5vdGlmaWNhdGlvblRocmVhZDIxMDYwMjc3MDE6NTg1MTI4MA%3D%3D#issuecomment-869565128. Closing the PR.

**Pull request closed without merging**